file.go

```diff
@@ -136,8 +137,22 @@ func (f *File) WriteTo(w io.Writer, opts ...Options) (int64, error) {
 // WriteToBuffer provides a function to get bytes.Buffer from the saved file,
 // and it allocates space in memory. Be careful when the file size is large.
 func (f *File) WriteToBuffer() (*bytes.Buffer, error) {
+	return f.WriteToBufferCustomZIP(func(w io.Writer) ZipWriter { return zip.NewWriter(w) })
+}
+
+type ZipWriter interface {
+	Create(name string) (io.Writer, error)
+	AddFS(fsys fs.FS) error
+	Close() error
+}
+
+// WriteToBufferCustomZIP provides a function to write to a buffer using a custom ZIP writer
+// factory. This allows for custom ZIP compression implementations to be used when saving
+// the spreadsheet to memory. The factory function should return a zipWriter interface
+// that will be used to create the final ZIP archive.
+func (f *File) WriteToBufferCustomZIP(factory func(io.Writer) ZipWriter) (*bytes.Buffer, error) {
```

Comment:
The `Save` and `SaveAs` function also should be support save workbook with custom zip writer. Introduce new function `SetZipWriter` just like `CharsetTranscoder` instead of this would be better approach.